src/nixos/machines/tupac/config/disks.nix

```diff
@@ -0,0 +1,236 @@
+											"--label=CRYPT_NIX"
+										];
+
+										extraOpenArgs = [
+											"--timeout 10"
+										];
+
+										content = {
+											type = "btrfs";
+											extraArgs = [ "--label NIX_STORE" ];
+											subvolumes = {
+												"@nix" = {
+													mountpoint = "/nix";
+													mountOptions = [ "compress=lzo" "noatime" ];
+												};
+												"@persist" = {
+													mountpoint = "/nix/persist/system";
+													mountOptions = [ "compress=lzo" "noatime" ];
+												};
+											};
+										};
+									};
+								};
+
+								swap = {
+									priority = 2;
+									size = "60G";
+									content = {
+										name = "swap";
+										type = "luks";
+
+										settings.allowDiscards = true;
+
+										passwordFile = config.age.secrets.tupac-disks-password.path;
+
+										initrdUnlock = true; # Add a boot.initrd.luks.devices entry for the specified disk
+
+										extraFormatArgs = [
+											"--use-random" # use true random data from /dev/random, will block until enough entropy is available
+											"--label=CRYPT_SWAP"
+										];
+
+										extraOpenArgs = [
+											"--timeout 10"
+										];
+
+										content = {
+											# FIXME-QA(Krey): Add label 'SWAP'
+											type = "swap";
+											resumeDevice = true; # resume from hiberation from this device
+
+											extraArgs = [
+												"--label SWAP"
+											];
+										};
+									};
+								};
+							};
+						};
+					};
+				};
+			};
+		} else {
+			disk = {
+				system = {
+					device = "/dev/disk/by-id/nvme-SOLIDIGM_SSDPFKNU010TZ_BTEH24220RNQ1P0B"; # NVME SSD
+					type = "disk";
+					content = {
+						type = "gpt";
+						partitions = {
+
+							boot = {
+								type = "EF00"; # EFI System Partition/
+								start = "2048";
+								end = "1050623"; # +512M
+								priority = 1; # Needs to be first partition
+								content = {
+									type = "filesystem";
+									format = "vfat"; # FAT32
+									mountpoint = "/boot";
+								};
+							};
+
+							root = {
+								start = "1050624";
+								end = "1874579455";
+								content = {
+									name = "root";
+									type = "luks";
+									settings.allowDiscards = true;
+
+									passwordFile = config.age.secrets.tupac-disks-password.path;
+
+									initrdUnlock = true; # Add a boot.initrd.luks.devices entry for the specified disk
+
+									extraFormatArgs = [
+										"--use-random" # use true random data from /dev/random, will block until enough entropy is available
+										"--label=CRYPT_NIXOS"
+									];
+
+									extraOpenArgs = [
+										"--timeout 10"
+									];
+
+									content = {
+										type = "btrfs";
+										extraArgs = [ "--label ROOT_NIXOS" ];
+										subvolumes = {
+											"@" = {
+												mountpoint = "/";
+												mountOptions = [ "compress=lzo" "noatime" ];
```

Comment:
I don't think the tradeoff of minor performance efficiency is worth the lack of access time date.. noatime should probably also be avoided, but we can't really touch the system's disk management until we fix https://github.com/NiXium-org/NiXium/issues/16 so that we can set user files as declarative through cloud sync